js/static/audioProcessor.js

```diff
@@ -0,0 +1,38 @@
+class AudioDataExtractor extends AudioWorkletProcessor {
```

Comment:
@Pierrci regarding[ this comment](https://github.com/huggingface/huggingface_hub/pull/738#issuecomment-1069668498), i'm using js rather than ts because https://github.com/microsoft/TypeScript/issues/28308

Reply:
did you try adding https://www.npmjs.com/package/@types/audioworklet?

Reply:
I've added [@types/audioworklet](https://www.npmjs.com/package/@types/audioworklet) and the following issues I'm facing are:
1. when it becomes `js/static/audioProcessor.ts`, it does not get transpiled to js (looking into why, possible error in [adapter-static](https://github.com/huggingface/hub-docs/blob/33b0b76b654c02a7755feeea837a9c2f05538615/js/svelte.config.js#L2))
2. to solve number 1, I've moved `js/static/audioProcessor.ts` -> `js/src/lib/components/InferenceWidget/shared/WidgetRealtimeRecorder/audioProcessor.ts`. In this case, I'm getting error `ReferenceError: AudioWorkletProcessor is not defined` (probably due to SSR)
3. to solve number 2, I've put the entire context of `js/static/audioProcessor.js` inside [WidgetRealtimeRecorder.svelte](https://github.com/huggingface/hub-docs/blob/1ec25435974979793189dae4cb586e4bae5941ea/js/src/lib/components/InferenceWidget/shared/WidgetRealtimeRecorder/WidgetRealtimeRecorder.svelte) `onMount` function. Still got `ReferenceError: AudioWorkletProcessor is not defined` inside the browser log.

I might be missing something here. Please let me know if you have any ideas

Reply:
Yes ok, this is 💩

The `ReferenceError` you were getting was probably because you're were trying to import the audio processor outside of the `audioWorklet.addModule` fn (`AudioWorkletProcessor` is only available inside the [`AudioWorkletGlobalScope`](https://developer.mozilla.org/en-US/docs/Web/API/AudioWorkletGlobalScope))

When `audioProcessor.ts` is moved in the `/shared` dir, using [imports as URL](https://github.com/vitejs/vite/blob/09c4fc01a83b84f77b7292abcfe7500f0e948db6/docs/guide/assets.md#explicit-url-imports) works correctly w/ the static adapter, but it doesn't work in moon-landing (obviously).

So I think we can keep it in `/static` (and as `.js`), and for it to work in `moon-landing` we will need to add an additional express static handler pointing to this file (which I'm not a fan of, but I don't see any other alternative?)

Reply:
yes, indeed, we do need that `/static` handler on moon-landing